File: leetcode/1647-minimum-deletions-to-make-character-frequencies-unique/solution.cpp
#include <string>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int minDeletions(string s) {
        int freq[26] = {0};
        for (char c : s) {
            freq[c - 'a']++;
        }

        unordered_set<int> used;
        int deletions = 0;

        for (int count : freq) {
            while (count > 0 && used.count(count)) {
                count--;
                deletions++;
            }
            if (count > 0) {
                used.insert(count);
            }
        }

        return deletions;
    }
};

File: leetcode/1647-minimum-deletions-to-make-character-frequencies-unique/solution.py
from collections import Counter


class Solution:
    def minDeletions(self, s: str) -> int:
        freq = Counter(s)
        used = set()
        deletions = 0

        for count in freq.values():
            while count > 0 and count in used:
                count -= 1
                deletions += 1
            if count > 0:
                used.add(count)

        return deletions
